utils/build-script

```diff
@@ -1203,69 +1205,21 @@ details of the setups of other systems or automated environments.""")
-        cmark_build_dir_label = args.cmark_build_variant
-        if args.cmark_assertions:
-            cmark_build_dir_label += "Assert"
-
-        llvm_build_dir_label = args.llvm_build_variant
-        if args.llvm_assertions:
-            llvm_build_dir_label += "Assert"
-
-        swift_build_dir_label = args.swift_build_variant
-        if args.swift_assertions:
-            swift_build_dir_label += "Assert"
-        if args.swift_analyze_code_coverage != "false":
-            swift_build_dir_label += "Coverage"
-
-        swift_stdlib_build_dir_label = args.swift_stdlib_build_variant
-        if args.swift_stdlib_assertions:
-            swift_stdlib_build_dir_label += "Assert"
-
-        # FIXME: mangle LLDB build configuration into the directory name.
-        if (llvm_build_dir_label == swift_build_dir_label and
-                llvm_build_dir_label == swift_stdlib_build_dir_label and
-                llvm_build_dir_label == cmark_build_dir_label):
-            # Use a simple directory name if all projects use the same build
-            # type.
-            args.build_subdir += "-" + llvm_build_dir_label
-        elif (llvm_build_dir_label != swift_build_dir_label and
-                llvm_build_dir_label == swift_stdlib_build_dir_label and
-                llvm_build_dir_label == cmark_build_dir_label):
-            # Swift build type differs.
-            args.build_subdir += "-" + llvm_build_dir_label
-            args.build_subdir += "+swift-" + swift_build_dir_label
-        elif (llvm_build_dir_label == swift_build_dir_label and
-                llvm_build_dir_label != swift_stdlib_build_dir_label and
-                llvm_build_dir_label == cmark_build_dir_label):
-            # Swift stdlib build type differs.
-            args.build_subdir += "-" + llvm_build_dir_label
-            args.build_subdir += "+stdlib-" + swift_stdlib_build_dir_label
-        elif (llvm_build_dir_label == swift_build_dir_label and
-                llvm_build_dir_label == swift_stdlib_build_dir_label and
-                llvm_build_dir_label != cmark_build_dir_label):
-            # cmark build type differs.
-            args.build_subdir += "-" + llvm_build_dir_label
-            args.build_subdir += "+cmark-" + cmark_build_dir_label
-        else:
-            # We don't know how to create a short name, so just mangle in all
-            # the information.
-            args.build_subdir += "+cmark-" + cmark_build_dir_label
-            args.build_subdir += "+llvm-" + llvm_build_dir_label
-            args.build_subdir += "+swift-" + swift_build_dir_label
-            args.build_subdir += "+stdlib-" + swift_stdlib_build_dir_label
+        args.build_subdir = compute_build_subdir(args)
 
-    build_dir = os.path.join(SWIFT_BUILD_ROOT, args.build_subdir)
+    workspace = Workspace(
+        source_root=SWIFT_SOURCE_ROOT,
+        build_root=os.path.join(SWIFT_BUILD_ROOT, args.build_subdir))
```

Comment:
Currently the `--build-subdir` command line option specifies an absolute path.  Would this still work with concatenation here?